fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java

```diff
@@ -1448,10 +1440,7 @@ private void getOldAlterJobInfos(Database db, List<List<Comparable>> schemaChang
     public ShowResultSet process(List<AlterClause> alterClauses, Database db, OlapTable olapTable)
             throws UserException {
         // index id -> index schema
-        Map<Long, LinkedList<Column>> indexSchemaMap = new HashMap<>();
-        for (Map.Entry<Long, List<Column>> entry : olapTable.getIndexIdToSchema().entrySet()) {
-            indexSchemaMap.put(entry.getKey(), new LinkedList<>(entry.getValue()));
-        }
+        Map<Long, List<Column>> indexSchemaMap = olapTable.getIndexIdToSchema();
```

Comment:
Here are the major modifications.

Reply:
The list copy is necessary because the list will be modified in `checkAndAddColumn`.

Reply:
`OlapTable.getIndexIdToSchema()` returns the original schema.

Reply:
> getIndexIdToSchema

```
    public Map<Long, List<Column>> getIndexIdToSchema() {
        Map<Long, List<Column>> result = Maps.newHashMap();
        for (Map.Entry<Long, MaterializedIndexMeta> entry : indexIdToMeta.entrySet()) {
            result.put(entry.getKey(), entry.getValue().getSchema());
        }
        return result;
    }
```

Reply:
`map.put` copys reference, not value.